openedx/core/djangoapps/discussions/serializers.py

```diff
@@ -187,13 +199,11 @@ def to_internal_value(self, data: dict) -> dict:
         """
         Transform the *incoming* primitive data into a native value.
         """
-        payload = {
-            'context_key': data.get('course_key', ''),
-            'enabled': data.get('enabled', False),
+        payload = super().to_internal_value(data)
```

Comment:
This lets the standard machinery handle the regular fields and we can focus on the complex cases.